src/main/java/de/relimit/commons/markdown/blockelement/admonition/Admonition.java

```diff
@@ -0,0 +1,77 @@
+package de.relimit.commons.markdown.blockelement.admonition;
+
+import java.util.Optional;
+
+import de.relimit.commons.markdown.blockelement.BlockElement;
+import de.relimit.commons.markdown.blockelement.MarkedBlockElementNode;
+import de.relimit.commons.markdown.configuration.MarkdownSerializationOptions;
+
+/**
+ * @see <a href=
+ *      "https://squidfunk.github.io/mkdocs-material/reference/admonitions/">https://squidfunk.github.io/mkdocs-material/reference/admonitions/</a>
+ */
+public class Admonition extends MarkedBlockElementNode implements BlockElement {
+
+	public static final Type DEFAULT_TYPE = DefaultType.NOTE;
+
+	public static final Expansion DEFAULT_EXPANSION = Expansion.FIXED;
+
+	public static final Display DEFAULT_DISPLAY = Display.BLOCK;
+
+	private Type type;
+
+	private Expansion expansion;
+
+	private Display display;
+
+	@Override
+	protected BlockElement gateKeep(BlockElement element) {
+		if (elements.isEmpty() && !(element instanceof Title)) {
+			throw new IllegalArgumentException(
+					"The first block element added to an admonition must be the " + Title.class.getSimpleName() + ".");
+		}
+		return element;
+	}
+
+	public Type getType() {
+		return type != null ? type : DEFAULT_TYPE;
+	}
+
+	public void setType(Type type) {
+		this.type = type;
+		invalidateSerialized();
+	}
+
+	public Expansion getExpansion() {
+		return expansion != null ? expansion : DEFAULT_EXPANSION;
+	}
+
+	public void setExpansion(Expansion expansion) {
+		this.expansion = expansion;
+		invalidateSerialized();
+	}
+
+	public Display getDisplay() {
+		return display != null ? display : DEFAULT_DISPLAY;
+	}
+
+	public void setDisplay(Display display) {
+		this.display = display;
+		invalidateSerialized();
+	}
+
+	@Override
+	protected String getMarker(MarkdownSerializationOptions options) {
```

Comment:
I don´t really understand this method so I cannot say if everything is correct here.

Reply:
I created a unit test for `Admonition` which might clear things up.
https://github.com/relimitgmbh/markdown-builder/blob/47c3e0689dcf22018e1b4e3ef4dc22e5f144dbab/src/test/java/de/relimit/commons/markdown/blockelement/BlockElementTests.java#L41

Also have a look at the specs by MkDocs:
https://squidfunk.github.io/mkdocs-material/reference/admonitions/#changing-the-title